map_test.go

```diff
@@ -1091,38 +1103,50 @@ func TestMapIterate(t *testing.T) {
 	}
 	defer hash.Close()
 
-	if err := hash.Put("hello", uint32(21)); err != nil {
-		t.Fatal(err)
-	}
+	var (
+		key   string
+		value uint32
+		keys  []string
+		data  = []string{"hello", "world"}
+	)
 
-	if err := hash.Put("world", uint32(42)); err != nil {
-		t.Fatal(err)
+	for i, k := range data {
+		if err := hash.Put(k, uint32(i)); err != nil {
+			t.Fatal(err)
+		}
 	}
 
-	var key string
-	var value uint32
-	var keys []string
-
 	entries := hash.Iterate()
 	for entries.Next(&key, &value) {
 		keys = append(keys, key)
 	}
-
-	if err := entries.Err(); err != nil {
-		t.Fatal(err)
-	}
+	qt.Assert(t, qt.IsNil(entries.Err()))
 
 	sort.Strings(keys)
+	qt.Assert(t, qt.DeepEquals(keys, data))
+}
 
-	if n := len(keys); n != 2 {
-		t.Fatal("Expected to get 2 keys, have", n)
-	}
-	if keys[0] != "hello" {
-		t.Error("Expected index 0 to be hello, got", keys[0])
-	}
-	if keys[1] != "world" {
-		t.Error("Expected index 1 to be hello, got", keys[1])
+func TestIterateWrongMap(t *testing.T) {
```

Comment:
I'm not sure if I understand the point of this test. Isn't the idea to make MapIterator support queues and stacks transparently later on? Why do we need to test here that it _doesn't_ work?

Reply:
The support for queues and stacks will be provided by the `Map.Drain` API, not `Map.Iterate` -- I would also like to later introduce the `TestDrainWrongMap` to expect an error when calling `Map.Drain` with a wrong map (i.e. an array map that doesn't support `LookupAndDelete`) --

If this test seems redundant I can remove it. However, it could help clarifying what is the expected behavior against specific types of maps. Let me know :smiley:

Reply:
Sorry! That makes a lot of sense, haven't had the opportunity to look at the other patches. In that case, testing that Map.Iterate() never returns any data and always returns an error is perfect. :+1:

Reply:
Np, thanks for the feedback, a double-check is always appreciated :wink: